### PR TITLE
A Fix the TypeError whenever prune is called before update has been e…

### PR DESCRIPTION
**Pull request by weidongtoml**
Fix the TypeError whenever prune is called before update has been executed.

 Before worker.update has been called, self.last_active is None; and if
 worker.prune is called at this time, TypeError: unsupported operand type(s)
 for +: 'NoneType' and 'float' will be raised in worker.prune.

### CODE REVIEW

**Comment by erikbern** [comment]
LGTM!

**Comment by Tarrasch** [comment]
Can you fix the pep8 issuse?

**Comment by Tarrasch** [comment]
Hmm, this haven't happened to us. Can you explain more as a user scenario when this happens?

**Comment by weidongtoml** [comment]
After the client had request the schedule server to add the task, due to some reason, the client was killed, so the server did not receive any ping, therefore the worker.upate method would never be called. Whenever this worker's prune method is invoked, it will result in the above exception.

**Comment by Tarrasch** [comment]
Ok. Great! But I would really like to see a test case added for this so we don't reintroduce the bug. Can you add that? :)

**Comment by weidongtoml** [comment]
Ok, will try to do that over the weekend. Btw, this is my first time to contribute so I am not sure what you meant by the pep8 issue,  and the Travis CI build failed mentioned here is it due to unit test failure?

**Comment by Tarrasch** [comment]
pep8 is just checking linting (like too many or few newlines). The actual tests did pass. :)

Here's a nice document mentioning how to test luigi locally: https://github.com/spotify/luigi/blob/master/CONTRIBUTING.rst :)

File: test/scheduler_test.py [hunk 1]
```diff
@@ -98,5 +98,14 @@ def reload_from_disk(cps):
             cps = reload_from_disk(cps=cps)  # This time without deleting
             self.assertEqual(cps.get_work(worker='D')['task_id'], '4')
 
+    def test_worker(self):
```
Comment (Tarrasch) [inline test/scheduler_test.py]: This name is not that descriptive. Can you rename it a bit :)?

**Comment by Tarrasch** [comment]
Thanks! Can you just squash your commits? This small change don't warrant 5 commits. :)

**Comment by weidongtoml** [comment]
Ok, have squashed them into one.

**Comment by Tarrasch** [comment]
Thanks @weidongtoml ! :)

File: luigi/scheduler.py [hunk 2]
```diff
@@ -224,7 +224,7 @@ class Worker(object):
     Structure for tracking worker activity and keeping their references.
     """
 
-    def __init__(self, worker_id, last_active=None):
+    def __init__(self, worker_id, last_active=time.time()):
```
Comment (erikbern) [inline luigi/scheduler.py]: This doesn't look right to me – should be `last_active=None` and then
```python
if last_active is None:
    last_active = time.time()
```

Comment (Tarrasch) [inline luigi/scheduler.py]: oh yea it is at module initialization time! Not worker creation time. But wait, if we just want to fix the type error, it might as well be `last_active=0.0`. But then `prune()` would kill the worker prematurely?

@weidongtoml, your thoughts?

Comment (weidongtoml) [inline luigi/scheduler.py]: yes, erik is right. Sorry, my mistake, I was thinking __init__(self, worker, last_active=time.time()) would initialise last_active at object creation time. Shall I submit another pull request for a fix?
